warehouse/query-core/src/main/java/datawave/query/index/lookup/ShardRangeStream.java

```diff
@@ -44,59 +44,59 @@ public CloseableIterable<QueryPlan> streamPlans(JexlNode node) {
             
             int stackStart = config.getBaseIteratorPriority() + 40;
-            BatchScanner scanner = scanners.newScanner(config.getShardTableName(), config.getAuthorizations(), config.getNumQueryThreads(), config.getQuery(),
-                            true);
-            
-            IteratorSetting cfg = new IteratorSetting(stackStart++, "query", FieldIndexOnlyQueryIterator.class);
-            
-            DefaultQueryPlanner.addOption(cfg, QueryOptions.QUERY_ID, config.getQuery().getId().toString(), false);
-            DefaultQueryPlanner.addOption(cfg, QueryOptions.QUERY, queryString, false);
-            
-            try {
-                DefaultQueryPlanner.addOption(cfg, QueryOptions.INDEX_ONLY_FIELDS,
-                                QueryOptions.buildFieldStringFromSet(metadataHelper.getIndexOnlyFields(config.getDatatypeFilter())), true);
-                DefaultQueryPlanner.addOption(cfg, QueryOptions.INDEXED_FIELDS,
-                                QueryOptions.buildFieldStringFromSet(metadataHelper.getIndexedFields(config.getDatatypeFilter())), true);
-            } catch (TableNotFoundException e) {
-                throw new RuntimeException(e);
-            }
-            
-            DefaultQueryPlanner.addOption(cfg, QueryOptions.START_TIME, Long.toString(config.getBeginDate().getTime()), false);
-            DefaultQueryPlanner.addOption(cfg, QueryOptions.DATATYPE_FILTER, config.getDatatypeFilterAsString(), false);
-            DefaultQueryPlanner.addOption(cfg, QueryOptions.END_TIME, Long.toString(config.getEndDate().getTime()), false);
-            
-            DefaultQueryPlanner.configureTypeMappings(config, cfg, metadataHelper, true);
-            
-            scanner.setRanges(Collections.singleton(rangeForTerm(null, null, config)));
-            
-            scanner.addScanIterator(cfg);
-            
-            Iterator<Entry<Key,Value>> kvIter = scanner.iterator();
-            
-            itr = Collections.emptyIterator();
-            
-            if (kvIter.hasNext()) {
-                PeekingIterator<Entry<Key,Value>> peeking = new PeekingIterator<>(kvIter);
-                Entry<Key,Value> peekKey = peeking.peek();
-                ErrorKey errorKey = ErrorKey.getErrorKey(peekKey.getKey());
-                if (errorKey != null) {
-                    switch (errorKey.getErrorType()) {
-                        case UNINDEXED_FIELD:
-                            this.context = StreamContext.UNINDEXED;
-                            break;
-                        case UNKNOWN:
-                            this.context = StreamContext.ABSENT;
-                    }
-                } else {
-                    itr = Iterators.transform(peeking, new FieldIndexParser(node));
-                    this.context = StreamContext.PRESENT;
+            try (BatchScanner scanner = scanners.newScanner(config.getShardTableName(), config.getAuthorizations(), config.getNumQueryThreads(),
```

Comment:
Similar to my previous comment, if we use try-with-resources then the scanner will be closed, invalidating the iterator which is used outside this method.  I don't think we want to do this.

Reply:
reverted